homeassistant/components/stream/__init__.py

```diff
@@ -139,10 +141,25 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     # pylint: disable=import-outside-toplevel
     from .recorder import async_setup_recorder
 
+    # Make a connection to the external url to see if we have an HTTP/2 proxy
+    http2_proxy_found = False
+    async with httpx.AsyncClient(http2=True) as client:
+        try:
+            response = await client.get(get_url(hass, allow_internal=False))
```

Comment:
I was thinking this through again and for my particular setup, i have http/2 in front of my internal url --  but then I don't have hairpin nat so my external url is not accessible internally.  So, i'm just one data point but i am not sure this detection would work for me. I realize my case is not common.

Reply:
I think that is a fair point. My external URL can only access `/api/camera_proxy` and `/api/webhook`. I primarily use my Internal URL for access it has the HTTP/2 proxy, the external one does not since it is only used for notifications and the like. I do not think it is safe to rely on specifically internal or external for this. 

Maybe either check both and then set them independently or allow the user to configure it?

Reply:
Thanks for the input. Given the wide range of different setups, I think we should just scrap this PR altogether. We can let users set the LL-HLS option themselves and perhaps default it to true at some point in the future.